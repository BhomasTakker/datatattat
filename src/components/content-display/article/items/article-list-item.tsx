import { CollectionItem } from "@/src/types/data-structures/collection/item/item";
import { ListItemContextProvider } from "./context/list-item.context";
import { ListItem, ListItemAvatar, ListItemText, Stack } from "@mui/material";
import { ArticleAvatar } from "./avatar/avatar";
import { Description, DescriptionVariants } from "./description/description";
import { DetailsComponent, DetailsVariantType } from "./details/details";
import {
	ContentTitle,
	ContentTitleVariants,
} from "./content-title/content-title";

export interface ArticleListItemProps {
	item: CollectionItem;
	titleVariant?: ContentTitleVariants;
	titleMaxLines?: number;
	descriptionMaxLines?: number;
	useAvatar: boolean;
	showDescription: boolean;
	descriptionVariant?: DescriptionVariants;

	detailsVariant?: DetailsVariantType;
	showPublished: boolean;
	showAuthor: boolean;
	showPublisher: boolean;
}

export const ArticleListItem = ({
	item,
	titleVariant = "Primary",
	titleMaxLines = 2,
	descriptionMaxLines = 3,
	useAvatar,
	showDescription,
	descriptionVariant = "Primary",
	detailsVariant = "space-between",
	showPublished,
	showAuthor,
	showPublisher,
}: ArticleListItemProps) => {
	const { title, avatar, src, description, guid, variant, details } = item;

	// const maxLines = 3;

	// hack for BING article
	// need update BING
	// or update sky / general RSS / chase this up and fix
	const img = avatar ? avatar.src : "";

	return (
		<ListItemContextProvider value={{ item }}>
			<ListItem data-testid="article-list-item">
				{/* if */}
				{useAvatar && (
					<ListItemAvatar>
						{/* Avatar Variant */}
						<ArticleAvatar alt={title} img={img} src={src} />
						{/* Icon */}
						{/* Logo */}
					</ListItemAvatar>
				)}
				<ListItemText
					disableTypography
					// title and title variant - == 'none'
					// always title? / we want type
					primary={<ContentTitle title={title} maxLines={titleMaxLines} />}
					secondary={
						// Create / get elsewhere
						<Stack margin={0} padding={0}>
							{/* if show */}
							{showDescription && (
								<Description
									description={description}
									maxLines={descriptionMaxLines}
								/>
							)}
							{(showAuthor || showPublished || showPublisher) && (
								<DetailsComponent
									details={details}
									showAuthors={showAuthor}
									showPublished={showPublished}
									showpublishers={showPublisher}
									showCategories={false}
									variant={detailsVariant}
								/>
							)}
						</Stack>
					}
				/>
			</ListItem>
		</ListItemContextProvider>
	);
};
