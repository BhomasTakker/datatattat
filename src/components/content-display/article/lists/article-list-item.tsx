import { CollectionItem } from "@/src/types/data-structures/collection/item/item";
import { ListItemContextProvider } from "../items/context/list-item.context";
import {
	ListItem,
	ListItemAvatar,
	ListItemText,
	Stack,
	Typography,
} from "@mui/material";
import { ArticleAvatar } from "../items/avatar/avatar";
import { Title } from "@/src/components/ui/title";
import { TitleVariant } from "@/src/components/types/ui";
import { stripHTML } from "@/src/utils/html";
import { Details } from "@/src/types/data-structures/base";
import { Time } from "../items/time/Time";
import { MARGINS } from "config/styles/styles.config";

const Description = ({ description }: { description: string }) => {
	console.log("FETURE:0007", "Description", { description });
	return (
		<Typography
			sx={{
				maxWidth: "100%",
				display: "-webkit-box",
				// function to return this data
				// take num lines
				overflow: "hidden",
				textOverflow: "ellipsis",
				WebkitBoxOrient: "vertical",
				WebkitLineClamp: 3,
				maxLines: 3,
			}}
			component="span"
			variant="body1"
			color="text.primary"
		>
			{stripHTML(description)}
		</Typography>
	);
};
const DetailsComponent = ({
	details,
	showPublished = false,
	showCategories = false,
	showpublishers = false,
	showAuthors = false,
}: {
	details?: Details;
	showPublished?: boolean;
	showCategories?: boolean;
	showpublishers?: boolean;
	showAuthors?: boolean;
}) => {
	const {
		published = null,
		categories = null,
		publishers = null,
		authors = null,
	} = details || {};
	console.log("FETURE:0007", "Details", {
		details,
		published,
		categories,
		publishers,
		authors,
	});
	return (
		<Stack
			direction="row"
			justifyContent={"space-between"}
			marginTop={MARGINS.SMALL}
		>
			{/* do something proper with the time!! */}
			{/* Like we want an image AND avatar component - we want Time - variant age/publish date */}
			{showPublished && <Time time={published} variant="age" />}
			{showCategories && <p>{categories?.join(", ")}</p>}
			{showpublishers && <p>{publishers?.join(", ")}</p>}
			{showAuthors && <p>{authors?.join(", ")}</p>}
		</Stack>
	);
};

export interface ArticleListItemProps {
	item: CollectionItem;
	useAvatar: boolean;
	showDescription: boolean;
	showPublished: boolean;
	showAuthor: boolean;
	showPublisher: boolean;
}

export const ArticleListItem = ({
	item,
	useAvatar,
	showDescription,
	showPublished,
	showAuthor,
	showPublisher,
}: ArticleListItemProps) => {
	const { title, avatar, src, description, guid, variant, details, media } =
		item;
	// hack for BING article
	// need update BING
	// or update sky / general RSS / chase this up and fix
	const img = avatar ? avatar.src : "";

	console.log("FETURE:0007", "ARTICLE:LIST:ITEM", {
		item,
		useAvatar,
		showDescription,
		showPublished,
		showAuthor,
		showPublisher,
	});

	return (
		<ListItemContextProvider value={{ item }}>
			<ListItem>
				{/* if */}
				{useAvatar && (
					<ListItemAvatar>
						<ArticleAvatar alt={title} img={img} src={src} />
						{/* Icon */}
						{/* Logo */}
					</ListItemAvatar>
				)}
				<ListItemText
					disableTypography
					// always title
					primary={<Title text={title} variant={TitleVariant.ARTICLE} />}
					secondary={
						<Stack>
							{/* if show */}
							{showDescription && <Description description={description} />}
							<DetailsComponent
								details={details}
								showAuthors={showAuthor}
								showPublished={showPublished}
								showpublishers={showPublisher}
								showCategories={false}
							/>
						</Stack>
					}
				/>
			</ListItem>
		</ListItemContextProvider>
	);
};
