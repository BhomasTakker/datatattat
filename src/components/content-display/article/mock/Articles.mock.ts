import {
	TIME_AGO_1_HOUR,
	TIME_AGO_1_WEEK,
	TIME_AGO_5_DAY,
} from "mockData/time";

export const Article1 = {
	title: "Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	src: "https://news.sky.com/story/dwayne-johnson-belly-laughs-after-controversial-waxwork-unveiled-12991640",
	avatar: {
		src: "https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712",
		alt: "Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	},
	media: {},
	description:
		"A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.",
	guid: "1",
	variant: "article",
	details: {
		docs: [],
		categories: ["category1", "category2"],
		authors: ["Author1", "Author2"],
		publishers: ["Publisher1", "Publisher1"],
		published: TIME_AGO_5_DAY,
		modified: "",
	},
};

export const Article2 = {
	title:
		"Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	src: "https://news.sky.com/story/dwayne-johnson-belly-laughs-after-controversial-waxwork-unveiled-12991640",
	avatar: {
		src: "https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712",
		alt: "Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	},
	media: {},
	description:
		"A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.",
	guid: "1",
	variant: "article",
	details: {
		docs: [],
		categories: ["category1", "category2"],
		authors: ["Author1", "Author2"],
		publishers: ["Publisher1", "Publisher1"],
		published: TIME_AGO_1_HOUR,
		modified: "",
	},
};

export const Article3 = {
	title:
		"Also Not Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	src: "https://news.sky.com/story/dwayne-johnson-belly-laughs-after-controversial-waxwork-unveiled-12991640",
	avatar: {
		src: "https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712",
		alt: "Dwayne Johnson 'belly laughs' after controversial waxwork unveiled",
	},
	media: {},
	description:
		"A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.",
	guid: "1",
	variant: "article",
	details: {
		docs: [],
		categories: ["category1", "category2"],
		authors: ["Author1", "Author2"],
		publishers: ["Publisher1", "Publisher1"],
		published: TIME_AGO_1_WEEK,
		modified: "",
	},
};

export const Articles = {
	title: "Article List Title",
	src: "https://news.sky.com/",
	description: "Article List Title - Lorem Ipsum ",
	guid: "0001",
	variant: "collection",
	details: {},
	media: {},
	pagination: {
		results: 0,
	},
	// Duplicate keys need to remedy
	items: [
		Article1,
		Article2,
		Article3,
		Article1,
		Article2,
		Article3,
		Article1,
		Article2,
		Article3,
	],
};
