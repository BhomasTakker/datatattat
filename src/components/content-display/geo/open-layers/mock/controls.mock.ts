import { format } from "ol/coordinate";
import { CreateControl } from "../controls/open-layers.controls";
import { createOpenLayersView } from "../view/open-layers.view";
import { getCurrentProjection } from "../projections/open-layers.projections";
import { createLayer } from "../layers/open-layers.layers";
import { getTileLayerSource } from "../layers/sources/open-layers.sources";

const view = createOpenLayersView({ projection: getCurrentProjection() });
const exampleLayer = createLayer(
	"TileLayer",
	{},
	getTileLayerSource("OSM", {})
);

export const mockOSControls: CreateControl[] = [
	{
		id: "FullScreen",
	},
	{
		// We want to associate id with relevant control
		id: "ZoomSlider",
		options: {
			duration: 1000,
		},
	},
	{
		id: "ZoomToExtent",
	},
	{
		id: "Zoom",
		options: { duration: 1000, delta: 10 },
	},
	{
		id: "Rotate",
		options: { autoHide: false },
	},
	{
		id: "Attributions",
		options: { collapsible: false, collapsed: false },
	},
	{
		id: "ScaleLine",
	},
	{
		id: "Overview",
		options: {
			collapsible: false,
			layers: [exampleLayer],
			view,
		},
	},
	{
		id: "MousePosition",
		options: {
			projection: "EPSG:4326",
			coordinateFormat: (coordinate) => {
				if (!coordinate) return "";

				return format(coordinate, "{y}, {x}", 3);
			},
		},
	},
];
