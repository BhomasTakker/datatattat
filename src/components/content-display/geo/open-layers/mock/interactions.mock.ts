import DragBox, { DragBoxEvent } from "ol/interaction/DragBox";
import { setSelectInteraction } from "../interactions/select";
import { SelectEvent } from "ol/interaction/Select";
import KML from "ol/format/KML";
import GeoJSON from "ol/format/GeoJSON";
import { DragAndDropEvent } from "ol/interaction/DragAndDrop";
import { altKeyOnly, altShiftKeysOnly } from "ol/events/condition";
import { CreateInteraction } from "../hooks/useMapInteractions";

// Okay this shit can be done!
const dragBoxCb = (e: DragBoxEvent) => {
	console.log({ e });

	const dragBoxInteraction = e.target as DragBox;
	const extent = dragBoxInteraction.getGeometry().getExtent();
	// const extent = e.coordinate;

	// const interactions = e.mapBrowserEvent.map.getInteractions();

	console.log({ extent });
	e.mapBrowserEvent.map.getView().fit(extent);
	// const extent = dragBoxInteraction.getGeometry().getExtent();
	// map.getView().fit(extent as Extent);
};

const selectInteraction = setSelectInteraction({
	cb: (selectEvent) => console.log({ selectEvent }),
});

const formatConstructors = [KML, GeoJSON];

// mocking / selecting these might be difficult?
// if something uses select say
// we are going to need to create that interaction as part of what uses it
export const mockOLInteractions: CreateInteraction[] = [
	{
		id: "DragBox",
		options: { minArea: 64, cb: dragBoxCb, condition: altShiftKeysOnly },
	},
	{
		id: "Select",
		options: { cb: (selectEvent: SelectEvent) => console.log({ selectEvent }) },
	},
	// {
	// 	id: "Modify",
	// 	options: { cb: (modifyEvent: ModifyEvent) => console.log({ modifyEvent }) },
	// },
	{
		id: "DragAndDrop",
		options: {
			formatConstructors,
			cb: (dragAndDropEvent: DragAndDropEvent) =>
				console.log({ dragAndDropEvent }),
		},
	},
];
