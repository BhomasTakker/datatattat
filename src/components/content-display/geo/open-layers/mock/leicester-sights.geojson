{
	"type": "FeatureCollection",
	"features": [
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [
					[
						[-1.1288642441902823, 52.66407147616076],
						[-1.1509241628855875, 52.66738934138226],
						[-1.1770002296757411, 52.65257550397584],
						[-1.189303725978192, 52.633633213459575],
						[-1.1887528231581541, 52.622598334817326],
						[-1.1876510263188038, 52.61951223479471],
						[-1.1904055345964935, 52.614599238597265],
						[-1.1843456035817894, 52.59508276744009],
						[-1.165064004899449, 52.5965328614667],
						[-1.1272353446268824, 52.59675594859587],
						[-1.0938139069099009, 52.606459139265354],
						[-1.093630272636858, 52.61649466094761],
						[-1.0758177481396842, 52.65572261534621],
						[-1.1079537459436324, 52.66764005144748],
						[-1.1288642441902823, 52.66407147616076]
					]
				],
				"type": "Polygon"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.1203762897064848, 52.62823007295199],
				"type": "Point"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.1111324536256575, 52.62663336950814],
				"type": "Point"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.1142300616747036, 52.62887172680553],
				"type": "Point"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.126802779062416, 52.62826984161606],
				"type": "Point"
			}
		},
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [-1.122353056064128, 52.627833204789965],
				"type": "Point"
			}
		}
	]
}
