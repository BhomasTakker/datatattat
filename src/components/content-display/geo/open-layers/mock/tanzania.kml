<kml xmlns="http://www.opengis.net/kml/2.2"><Document>
<Placemark id="0">
<name>Tanzania</name><ExtendedData>
  <Data name="rank"><value>100</value></Data></ExtendedData>
  <Polygon>
<outerBoundaryIs>
  <LinearRing><coordinates>32.75146220425401,-4.070616210654833
32.404539433405716,-4.416006505052337
31.68937101079902,-3.9930872097108363
31.367818343967855,-4.606212468389984
31.21849996178358,-5.685102878965196
32.04422842866629,-5.669933877989223
32.14082124830054,-6.105766138192507
32.71304648173975,-6.405460216769654
32.91941166834397,-7.534157867951166
33.028791094590275,-8.190802587342617
33.18975133271857,-8.463810569749057
33.434517126648245,-8.902060461160033
33.67062867767021,-8.973920016476143
34.13094567873361,-8.63879372210613
35.057459215494475,-8.9818497143238
35.3765135478655,-8.886282241240806
35.74365892026407,-8.92014039510751
36.15880565317889,-8.52425649123434
37.14866035700493,-7.629257872171195
36.598688650828166,-7.415940323361539
36.62531016269381,-6.761452542501388
36.80213245144557,-6.082806142322184
36.82466118482168,-5.813095013944661
38.052518323910704,-5.2183303501712714
38.16222817018428,-4.699502938804272
37.52959565262924,-3.7772372002447128
36.44168359502956,-3.6426337760581333
34.26285980384148,-3.981884004914292
32.75146220425401,-4.070616210654833</coordinates></LinearRing></outerBoundaryIs></Polygon></Placemark>
<Placemark>
<name>River Tanzania</name><ExtendedData>
  <Data name="rank"><value>80</value></Data></ExtendedData>
  <LineString><coordinates>27.436710640216916,-1.6114971461392429
28.933878301348386,-3.2324072110514095
31.087225715863553,-3.2861422919907994
32.09991956876732,-3.8530697553275104
32.97636792105874,-4.668701790983647
35.38679649868058,-5.2416479385965005
37.001848300138334,-5.894152767456376
37.945199955192976,-6.927904833995697
39.49355577917615,-7.138065470147907</coordinates></LineString></Placemark>
<Placemark>
<name>Leicester</name><ExtendedData>
  <Data name="rank"><value>90</value></Data></ExtendedData>
  <Point><coordinates>36.55268988479435,-4.255704272696974</coordinates></Point></Placemark>
<Placemark>
<name>Derby</name><ExtendedData>
  <Data name="rank"><value>70</value></Data></ExtendedData>
  <Point><coordinates>33.21689440655595,-5.71183141606754</coordinates></Point></Placemark>
<Placemark>
<name>Nottingham</name><ExtendedData>
  <Data name="rank"><value>50</value></Data></ExtendedData>
  <Point><coordinates>34.22044256874557,-7.528800702788601</coordinates></Point></Placemark>
<Placemark>
<name>Coventry</name><ExtendedData>
  <Data name="rank"><value>12</value></Data></ExtendedData>
  <Point><coordinates>35.784394588476346,-5.865783514020492</coordinates></Point></Placemark></Document></kml>