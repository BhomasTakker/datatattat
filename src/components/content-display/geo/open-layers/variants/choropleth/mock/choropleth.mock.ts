import { Feature } from "ol";
import { ColorMap, Filter } from "../../../filters/types";
import { ApplyProportionalColor } from "../../../style/apply-proportional-color";
import { transform } from "ol/proj";
import { Point, Polygon } from "ol/geom";

const colorMap: ColorMap = {
	filter: "equals",
	map: [
		{
			value: "Namibia",
			key: "name",
			color: [255, 0, 0, 0.7],
		},
		{
			value: "South Africa",
			key: "name",
			color: [0, 255, 0, 0.7],
		},
		{
			value: "Botswana",
			key: "name",
			color: [255, 0, 255, 0.7],
		},
		{
			value: "Zimbabwe",
			key: "name",
			color: [0, 0, 255, 0.7],
		},
		{
			value: "Madegascar",
			key: "name",
			color: [0, 255, 255, 0.7],
		},
		{
			value: "Mozambique",
			key: "name",
			color: [255, 255, 0, 0.7],
		},
	],
};

const proportionalColor: Omit<ApplyProportionalColor, "object"> = {
	key: "population_density",
	gradientStart: [77, 255, 255, 1],
	gradientEnd: [255, 255, 255, 1],
	minValue: 0,
	maxValue: 1500,
};
const filter: Filter = {
	filter: "lessThan",
	value: 1200,
	key: "population_density",
};
const filter2: Filter = {
	filter: "greaterThan",
	value: 500,
	key: "population_density",
};
const filters = [filter, filter2];

const attributes = {
	name: "Leicester",
	bar: "foo",
	wahwah: "woowoo",
	score: 0.6,
	rank: 75,
};

// Figure out or rem - points are far more likely
// Not working :()
const createFeature = () => {
	return new Feature({
		geometry: new Polygon([
			[
				[11.852132950632637, -17.17559188963027],
				[15.6884190829482, -28.498698388783247],
				[19.511230992915074, -28.85018047206215],
				[21.20061858837039, -17.803784140414393],
				[11.852132950632637, -17.17559188963027],
			],
		]),
		...attributes,
	});
};

export const ChoroplethMock = {
	filters,
	colorMap,
	proportionalColor,
	createFeature,
};
