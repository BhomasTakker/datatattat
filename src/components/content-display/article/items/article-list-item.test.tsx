import { render, screen } from "@testing-library/react";
import { ArticleListItem } from "./article-list-item";
import { Article1 } from "../lists/article-list.mock.data";

const props = {
	item: Article1,
	useAvatar: true,
	showDescription: true,
	showPublished: true,
	showAuthor: true,
	showPublisher: true,
};

describe("ArticleListItem", () => {
	it("should render article list item", () => {
		render(<ArticleListItem {...props} />);
		const articleListItem = screen.getByTestId("article-list-item");
		expect(articleListItem).toBeInTheDocument();
	});

	describe("snaps", () => {
		it("should match snapshot for article list item", () => {
			render(<ArticleListItem {...props} />);
			const articleListItem = screen.getByTestId("article-list-item");
			expect(articleListItem).toMatchSnapshot();
		});
	});
});
