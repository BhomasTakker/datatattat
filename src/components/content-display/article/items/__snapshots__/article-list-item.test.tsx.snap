// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem snaps should match snapshot for article list item 1`] = `
<li
  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
  data-testid="article-list-item"
>
  <div
    class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
      data-testid="avatar"
    >
      <img
        alt="Dwayne Johnson 'belly laughs' after controversial waxwork unveiled"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="https://e3.365dm.com/23/10/70x70/skynews-dwayne-johnson-waxwork_6333809.jpg?20231024132712"
      />
    </div>
  </div>
  <div
    class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-mkqqcn-MuiTypography-root"
    >
      Dwayne Johnson 'belly laughs' after controversial waxwork unveiled
    </h6>
    <div
      class="css-urd5lq-MuiStack-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 css-1aunven-MuiTypography-root"
        data-testid="description"
      >
        A museum has changed the skin tone of its Dwayne Johnson wax statue after it was criticised for being too white.
      </span>
      <div
        class="css-1jligzp-MuiStack-root"
        data-testid="details"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-sbmvo9-MuiTypography-root"
          data-testid="time"
        >
          Published 5 days ago
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-aiw75n-MuiTypography-root"
        >
          Publisher1, Publisher1
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-aiw75n-MuiTypography-root"
        >
          Author1, Author2
        </p>
      </div>
    </div>
  </div>
</li>
`;
